spec/function.dd

```diff
@@ -2884,10 +2884,60 @@ $(H3 $(LNAME2 safe-functions, Safe Functions))
         so they can be corrected.
         )
 
+$(H4 Safe External Functions)
+
+        $(P External functions don't have a function body visible to the compiler:
+        ---
+        @safe extern (C) void play();
+        ---
+        and so safety cannot be verified automatically.
+        )
+
+        $(BEST_PRACTICE Explicitly set an attribute for external functions rather
+        than relying on default settings.)
+
 $(H3 $(LNAME2 trusted-functions, Trusted Functions))
 
         $(P Trusted functions are marked with the $(CODE @trusted) attribute.)
 
+        $(P Trusted functions act as an interface between safe and system functions.
+        A trusted function provides a safe interface. A $(I Safe Interface) is one that:
+
+        $(UL
+
+        $(LI Does not invalidate pointers that are passed to it. For example,
+        ---
+        @system void free(void* p);
+        ...
+        free(p);
+        *p = 3;  // p is invalid
+        ---
+        is not a safe interface because `free(p)` means `p` is no longer a valid
+        pointer when it returns. `free` can only be `@system`.)
+
+        $(LI Does not attempt to access memory beyond the bounds of the memory object
+        a parameter points to. For example:
+        ---
+        @system size_t strlen(char* p);
+        ---
+        is not a safe interface because the implementation of `strlen` advances `p` until
+        it finds a `null` character, and this can be arbitrarily beyond the `char` that `p`
+        purportedly points to. `strlen` can only be `@system`. Any function that traverses
+        a C string passed as an argument can only be `@system`.
+        )
+
+        $(LI Has array bounds passed as a separate parameter rather than as part of the pointer
+        type. For example:
```

Comment:
This third item seems to be the opposite of what it should be. Shouldn't it be that array bounds as a separate parameter are unsafe, while the other is safe?

Reply:
good catch